am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/GetResourceDefinitionsWithCreatePermissionIntegrationTest.java

```diff
@@ -26,93 +27,98 @@
 import static uk.gov.hmcts.reform.amlib.helpers.DefaultRoleSetupDataFactory.createPermissionsForAttribute;
 import static uk.gov.hmcts.reform.amlib.helpers.DefaultRoleSetupDataFactory.createResourceDefinition;
-import static uk.gov.hmcts.reform.amlib.helpers.TestConstants.OTHER_ROLE_NAME;
-import static uk.gov.hmcts.reform.amlib.helpers.TestConstants.ROLE_NAME;
 import static uk.gov.hmcts.reform.amlib.helpers.TestConstants.ROOT_ATTRIBUTE;
 
-@SuppressWarnings({"PMD.TooManyMethods", "LineLength"})
+@SuppressWarnings("PMD.TooManyMethods")
 class GetResourceDefinitionsWithCreatePermissionIntegrationTest extends PreconfiguredIntegrationBaseTest {
     private static AccessManagementService service = initService(AccessManagementService.class);
     private static DefaultRoleSetupImportService importerService = initService(DefaultRoleSetupImportService.class);
     private ResourceDefinition resourceDefinition;
     private ResourceDefinition otherResource;
+    private String roleName;
+    private String otherRoleName;
 
     @BeforeEach
     void setUp() {
-        importerService.addResourceDefinition(
-            resourceDefinition = createResourceDefinition(serviceName, UUID.randomUUID().toString(), UUID.randomUUID().toString()));
-        importerService.addResourceDefinition(
-            otherResource = createResourceDefinition(serviceName, UUID.randomUUID().toString(), UUID.randomUUID().toString()));
+        roleName = UUID.randomUUID().toString();
+        otherRoleName = UUID.randomUUID().toString();
+        importerService.addRole(roleName, IDAM, PUBLIC, EXPLICIT);
+        importerService.addRole(otherRoleName, IDAM, PUBLIC, EXPLICIT);
+        importerService.addResourceDefinition(resourceDefinition =
+            createResourceDefinition(serviceName, UUID.randomUUID().toString(), UUID.randomUUID().toString()));
+        importerService.addResourceDefinition(otherResource =
+            createResourceDefinition(serviceName, UUID.randomUUID().toString(), UUID.randomUUID().toString()));
     }
 
     @Test
     void shouldRetrieveResourceDefinitionWhenRecordExists() {
-        importerService.grantDefaultPermission(createDefaultPermissionGrant(resourceDefinition, ROOT_ATTRIBUTE, ImmutableSet.of(CREATE)));
+        importerService.grantDefaultPermission(createDefaultPermissionGrant(
+            resourceDefinition, ImmutableSet.of(CREATE), "", roleName));
 
         Set<ResourceDefinition> result =
-            service.getResourceDefinitionsWithRootCreatePermission(ImmutableSet.of(ROLE_NAME));
+            service.getResourceDefinitionsWithRootCreatePermission(ImmutableSet.of(roleName));
 
         assertThat(result).containsExactly(resourceDefinition);
     }
 
     @Test
     void shouldRetrieveResourceDefinitionWhenRecordExistsWithMultiplePermissions() {
-        Set<Permission> permissions = ImmutableSet.of(READ, CREATE);
+        importerService.grantDefaultPermission(createDefaultPermissionGrant(
```

Comment:
Why attribute is passed after permissions? Usually attributes comes before.

Would changing it trigger much more changes? Just asking before you start making extra changes.

Reply:
👍